### PR TITLE
log-backup: add progress bar for truncate

### LINKED ISSUE #36914: log-backup: add progress bar for truncating
## Enhancement

As title.

### PR DESCRIPTION
**Pull request by YuJuncen**
### What problem does this PR solve?

Issue Number: close #36914

Problem Summary:
`br log truncate` may keep for a long time, however for now there isn't anything indices the progress.

### What is changed and how it works?
Added a progress bar using `mpb`. Also a new interface in the `ConsoleOps` have been added.

image

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
This script can generate the progress bar in last segment.
```go
package main

import (
	"time"

	"github.com/pingcap/tidb/br/pkg/glue"
)

func main() {
	g := glue.ConsoleOperations{glue.StdIOGlue{}}
	total := 10000

	pb := g.StartProgressBar("Test", total, glue.WithTimeCost(), glue.WithConstExtraField("hello", "world"))
	for i := 0; i < total; i++ {
		pb.Inc()
		time.Sleep(10 * time.Millisecond)
	}

	pb.Wait()

}
```
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
`br log truncate` now have a progress bar.
```

### CODE REVIEW

File: br/pkg/task/stream.go [hunk 1]
```diff
@@ -922,7 +922,7 @@ func RunStreamTruncate(c context.Context, g glue.Glue, cmdName string, cfg *Stre
 	removed := metas.RemoveDataBefore(shiftUntilTS)
 
 	// remove metadata
-	removeMetaDone := console.ShowTask("Removing metadata... ", glue.WithTimeCost())
+	removeMetaDone := console.ShowTask("Removing Metadata... ", glue.WithTimeCost())
 	if !cfg.DryRun {
 		if err := metas.DoWriteBack(ctx, storage); err != nil {
```
Comment (MoCuishle28) [inline br/pkg/task/stream.go:927]: Currently, the log file is removed first, and then the meta file is removed or written back. Shall we put `metas.DoWriteBack` into Progress? And call `p.Wait()` after `removeMetaDone()`.
e.g.,
```
metas.DoWriteBack(ctx, storage, func() {
    p.Inc()
})
```

File: br/pkg/task/stream.go [hunk 2]
```diff
@@ -941,38 +941,65 @@ func RunStreamTruncate(c context.Context, g glue.Glue, cmdName string, cfg *Stre
...
+			if cfg.DryRun {
+				return
+			}
+			if err := storage.DeleteFile(ctx, f.Path); err != nil {
+				log.Warn("File not deleted.", zap.String("path", f.Path), logutil.ShortError(err))
+				notDeleted = append(notDeleted, f.Path)
+			}
+		})
+	}
+	if err := p.Wait(ctx); err != nil {
```
Comment (3pointer) [inline br/pkg/task/stream.go:973]: can we wait all worker finish apply here?

Comment (YuJuncen) [reply br/pkg/task/stream.go:973]: If the total count is `0`, the progress would finish as soon as it being created.

**Review by 3pointer — APPROVED** [review]

File: br/pkg/task/stream.go [hunk 3]
```diff
@@ -941,38 +944,81 @@ func RunStreamTruncate(c context.Context, g glue.Glue, cmdName string, cfg *Stre
...
+	eg, cx := errgroup.WithContext(ctx)
+	const keepFirstNFailure = 16
+	var notDeleted struct {
+		item []string
+		sync.Mutex
+	}
 	for _, f := range removed {
-		if !cfg.DryRun {
-			wg.Add(1)
-			finalFile := f
```
[the remark this answers is not in the document]
Comment (Leavrth) [reply br/pkg/task/stream.go:954]: `finalFile := f`

File: br/pkg/task/stream.go [hunk 4]
```diff
@@ -941,38 +944,81 @@ func RunStreamTruncate(c context.Context, g glue.Glue, cmdName string, cfg *Stre
...
+	notDeleted.Unlock()
 
 	// remove metadata
-	removeMetaDone := console.ShowTask("Removing metadata... ", glue.WithTimeCost())
-	if !cfg.DryRun {
-		if err := metas.DoWriteBack(ctx, storage); err != nil {
-			return err
-		}
+	pw := console.StartProgressBar("Removing Metadata", metas.PendingMeta(), glue.WithTimeCost(), glue.WithConstExtraField("metas", metas.PendingMeta()))
+	defer pw.Close()
```
Comment (Leavrth) [inline br/pkg/task/stream.go:1010]: Does package `mpb` support to close the second progress bar `pw` earlier than `p`?

Comment (YuJuncen) [reply br/pkg/task/stream.go:1010]: pw would always be open after `p` closed. 🤔

Comment (YuJuncen) [reply br/pkg/task/stream.go:1010]: (Note that `mpb` would stop refreshing once it has been fulfilled.)

File: br/pkg/task/stream.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (YuJuncen) [reply br/pkg/task/stream.go:954]: I think perhaps we actually need to add some unit tests for `truncate`... 🥲

File: br/pkg/task/stream.go (hunk 4 above)
Comment (Leavrth) [reply br/pkg/task/stream.go:1010]: get

**Pull request merged**